### PR TITLE
feat(api-clients/grpc): add variables argument for FailJob gRPC

### LINKED ISSUE #1387: I can set variables when failing a job
As a user, I can set variables when failing a job. (issue: [camunda/zeebe#9404](https://github.com/camunda/zeebe/issues/9404), pr: [https://github.com/camunda/zeebe/pull/10702](https://github.com/camunda/zeebe/pull/10702) )

### PR DESCRIPTION
**Pull request by skayliu**
## What is the purpose of the change

Add variables argument for FailJob gRPC.

closes #1387 

## Are there related marketing activities

No.

## When should this change go live?

May be C8.2.

## PR Checklist

- [ ] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by skayliu** [comment]
Hi, @korthout, I've updated the doc, Please review this, Thank you.

File: docs/apis-clients/grpc.md [hunk 1]
```diff
@@ -395,7 +395,7 @@ Returned if:
 Marks the job as failed. If the retries argument is positive and no retry back off is set, the job is immediately
 activatable again. If the retry back off is positive the job becomes activatable once the back off timeout has passed.
 If the retries argument is zero or negative, an incident is raised, tagged with the given errorMessage, and the job is
-not activatable until the incident is resolved.
+not activatable until the incident is resolved. If the variables argument is set, the given valid JSON document will be set locally to the job element.
```
Comment (korthout) [inline docs/apis-clients/grpc.md:398]: 🔧 Your description is correct, but we could use some common terminology (like merging variables into the process)
```suggestion
not activatable until the incident is resolved. If the variables argument is set, the variables are merged into the process at the local scope of the job's associated task.
```

**Review by korthout — DISMISSED** [review]
Thanks @skayliu 🚀 

This change made me wonder whether we also need to update:
- https://docs.camunda.io/docs/next/components/concepts/job-workers/#completing-or-failing-jobs

❓ But I'm not sure, because that page explains the basic job worker concept while this feature is for rather advanced use cases. I'd be curious to hear your opinion about this.

🔧 Please consider my suggestion

**Comment by skayliu** [comment]
> * https://docs.camunda.io/docs/next/components/concepts/job-workers/#completing-or-failing-jobs

@korthout  what you think add these?

When `Completing or failing  jobs` with variables, the variables are merged into the process at the the job's associated task.

- `Completing a job` , When variables are merged into a process instance, the variables are  propagated from the scope of the activity to its higher scopes.
- `Failing a job`, When variables are merged into a process instance, the variables are only created in the local scope of the task.

**Review by christinaausley — APPROVED** [review]
No adjustments from a technical writing perspective on this one 👍

File: docs/components/concepts/job-workers.md [hunk 2]
```diff
@@ -72,6 +72,11 @@ This could be useful when a job worker communicates with an external system. If
 This will result in an incident when the retries run out. Using the `retry back off` will delay the retry. This allows the external system some time to recover.
 If no `retry back off` the job is immediately retried.
 
+When `Completing or failing jobs` with variables, the variables are merged into the process of the job's associated task.
```
Comment (korthout) [inline docs/components/concepts/job-workers.md:75]: When reading the Concepts documentation from start to finish, the Job Worker documentation comes before the Variables documentation. So readers are unlikely to be familiar with the Variables concept when they read this. 

🔧 I think we should link to it, so they can read more about Variables if they are interested.

```suggestion
When `Completing or failing jobs` with [variables](components/concepts/variables.md), the variables are merged into the process at the job's associated task.
```

File: docs/components/concepts/job-workers.md [hunk 3]
```diff
@@ -72,6 +72,11 @@ This could be useful when a job worker communicates with an external system. If
...
+
+- If `Completing a job` the variables are propagated from the scope of the task to its higher scopes.
+- If `Failing a job` the variables are only created in the local scope of the task.
```
Comment (korthout) [inline docs/components/concepts/job-workers.md:78]: 🔧 I'm not a native English speaker, but I think it should be:

```suggestion
- When `Completing a job` the variables are propagated from the scope of the task to its higher scopes.
- When `Failing a job` the variables are only created in the local scope of the task.
```

**Review by korthout — COMMENTED** [review]
Thanks for the additional changes @skayliu 

@christinaausley Let me know what you think about linking ahead in the concepts and about my grammar suggestion.

File: docs/components/concepts/job-workers.md (hunk 2 above)
Comment (christinaausley) [reply docs/components/concepts/job-workers.md:75]: 👍 Nice change here.

File: docs/components/concepts/job-workers.md (hunk 3 above)
Comment (christinaausley) [reply docs/components/concepts/job-workers.md:78]: Good catch 👍

**Pull request merged**